AngouriMath/Core/Sys/Entity.cs

```diff
@@ -304,47 +294,33 @@ public struct EntityTensorWrapperOperations : IOperations<Entity>
             /// If you need vector, list 1 dimension (length of the vector)
             /// You can't list 0 dimensions
             /// </summary>
-            /// <param name="dims"></param>
-            public Tensor(params int[] dims)
-                : this(GenTensor<Entity, EntityTensorWrapperOperations>.CreateTensor(new(dims), inds => 0)) { }
-
-            public Entity this[params int[] dims]
-            {
-                get => InnerTensor[dims];
-                set => InnerTensor[dims] = value;
-            }
+            public Tensor(Func<int[], Entity> operation, params int[] dims) : this(GenTensor.CreateTensor(new(dims), operation)) { }
 
+            public Entity this[int i] => InnerTensor[i];
+            public Entity this[int x, int y] => InnerTensor[x, y];
+            public Entity this[int x, int y, int z] => InnerTensor[x, y, z];
+            public Entity this[params int[] dims] => InnerTensor[dims];
             public bool IsVector => InnerTensor.IsVector;
             public bool IsMatrix => InnerTensor.IsMatrix;
 
-            /// <summary>
-            /// Changes the order of axes
-            /// </summary>
-            /// <param name="a"></param>
-            /// <param name="b"></param>
+            /// <summary>Changes the order of axes</summary>
             public void Transpose(int a, int b) => InnerTensor.Transpose(a, b);
 
-            /// <summary>
-            /// Changes the order of axes in matrix
-            /// </summary>
+            /// <summary>Changes the order of axes in matrix</summary>
             public void Transpose()
             {
-                if (IsMatrix)
-                    InnerTensor.TransposeMatrix();
-                else
-                    throw new Core.Exceptions.MathSException("Specify axes numbers for non-matrices");
+                if (IsMatrix) InnerTensor.TransposeMatrix();
+                else throw new Core.Exceptions.MathSException("Specify axes numbers for non-matrices");
             }
 
             // We do not need to use Gaussian elimination here
             // since we anyway get N! memory use
             public Entity Determinant() => InnerTensor.DeterminantLaplace();
 
-            /// <summary>
-            /// Inverts all matrices in a tensor
-            /// </summary>
+            /// <summary>Inverts all matrices in a tensor</summary>
             public Tensor Inverse()
             {
-                var cp = InnerTensor.Copy(copyElements: true);
+                var cp = InnerTensor.Copy(false);
```

Comment:
explicit naming good >:|. Anyway, minooor

Reply:
So when are explicit naming encouraged? All bool parameters? Better to set up an analyzer.